monkey/infection_monkey/tunnel.py

```diff
@@ -181,8 +182,9 @@ def run(self):
 
         # wait till all of the tunnel clients has been disconnected, or no one used the tunnel in
         # QUIT_TIMEOUT seconds
-        # TODO: Replace with infection_monkey.utils.timer.Timer
-        while self._clients and (time.time() - get_last_serve_time() < QUIT_TIMEOUT):
+        timer = Timer()
+        timer.set(self._calculate_timeout())
+        while self._clients and not timer.is_expired():
```

Comment:
As long as this is thoroughly tested, I approve. I don't think there are unit tests for it.